packages/flutter_goldens/lib/src/flutter_goldens_io.dart

```diff
@@ -20,28 +22,110 @@ export 'package:flutter_goldens_client/skia_client.dart';
 
 const String _kFlutterRootKey = 'FLUTTER_ROOT';
 
+/// Similar to [matchesGoldenFile] but specialized for Flutter's own tests.
+///
+/// For descriptions of [key] and [version] parameters see [matchesGoldenFile].
+///
+/// If [isFlaky] is false, the golden check fails if the screenshot generated
+/// by the test does not match the golden stored in Skia Gold.
+///
+/// If [isFlaky] is true, the golden check will always pass no matter how many
+/// pixels deviate from the golden, and no matter how big the color difference
+/// is per pixel.
+///
+/// Whether [isFlaky] is true or false, the generated golden is always uploaded
+/// to Skia Gold for manual inspection. Use [isFlaky] to skip a flaky golden,
+/// and monitor it using the Skia Gold UI, by visiting https://flutter-gold.skia.org/list,
+/// and clicking on the respective golden test name. The UI will show the
+/// history of generated goldens over time. Each unique golden gets a unique
+/// color. If the color is the same for all commits in the recent history, the
+/// golden is likely no longer flaky and [isFlaky] can be set back to false. If
+/// the color changes from commit to commit then it is still flaky.
+AsyncMatcher matchesFlutterGolden(Object key, { int? version, bool isFlaky = false }) {
+  if (key is Uri) {
+    return MatchesGoldenFile(key, version, comparator: isFlaky ? _flakyGoldenFileComparator : null);
+  } else if (key is String) {
+    return MatchesGoldenFile.forStringPath(key, version, comparator: isFlaky ? _flakyGoldenFileComparator : null);
+  }
+  throw ArgumentError('Unexpected type for golden file: ${key.runtimeType}');
+}
+
+String? _namePrefix;
+late GoldenComparatorType _flakyGoldenFileComparator;
+
 /// Main method that can be used in a `flutter_test_config.dart` file to set
 /// [goldenFileComparator] to an instance of [FlutterGoldenFileComparator] that
 /// works for the current test. _Which_ FlutterGoldenFileComparator is
 /// instantiated is based on the current testing environment.
 ///
 /// When set, the `namePrefix` is prepended to the names of all gold images.
 Future<void> testExecutable(FutureOr<void> Function() testMain, {String? namePrefix}) async {
+  _namePrefix = namePrefix;
+  final GoldenFileComparator nonFlakyComparator = await _createComparator();
+  _flakyGoldenFileComparator = await _createComparator(isFlaky: true);
+
+  // Assign this last because FlutterLocalFileComparator.fromDefaultComparator
```

Comment:
I am not sure I follow, is this a bug?

Reply:
I admit, I don't understand the golden design well enough to tell. The problematic line is https://github.com/flutter/flutter/blob/56c4e3cdd8a11ba373979e05f48836d1440e7d25/packages/flutter_goldens/lib/flutter_goldens.dart#L233

So if I assign `goldenFileComparator` to anything other than `LocalFileComparator` the `fromDefaultComparator` fails the `as` cast, so I'm deferring the assignment till after all invocations of `fromDefaultComparator`.

Reply:
I wonder if this could be a lot simpler if the actual comparator doesn't have anything to do with flakiness